#!/usr/bin/env python

# Copyright (C) 2017 Electric Movement Inc.
# All Rights Reserved.

# Author: Brandon Kinman
# Modified by Jun Zhu

PACKAGE = "attitude_controller"

import math
from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

max_angle = round(math.pi/2, 4)
gen.add('use_dr_set_point', bool_t,   0, 'Use dynamic reconfig setpoint',  False)
# it makes no sense to let roll or pitch larger than pi/2
gen.add('roll_set_point',   double_t, 0, 'Set Point: X', 0, -max_angle, max_angle)
gen.add('pitch_set_point',  double_t, 0, 'Set Point: Y', 0, -max_angle, max_angle)
gen.add('yaw_set_point',    double_t, 0, 'Set Point: Z', 0, -2*max_angle, 2*max_angle)

gen.add('roll_kp',     double_t, 0, 'Roll P',     0, 0, 100)
gen.add('roll_ki',     double_t, 0, 'Roll I',     0, 0, 100)
gen.add('roll_ki_max', double_t, 0, 'Roll I Max', 0, 0, 100)
gen.add('roll_kd',     double_t, 0, 'Roll D',     0, 0, 100)

gen.add('pitch_kp',     double_t, 0, 'Pitch P',     0, 0, 100)
gen.add('pitch_ki',     double_t, 0, 'Pitch I',     0, 0, 100)
gen.add('pitch_ki_max', double_t, 0, 'Pitch I Max', 0, 0, 100)
gen.add('pitch_kd',     double_t, 0, 'Pitch D',     0, 0, 100)

gen.add('yaw_kp',     double_t, 0, 'Yaw P',     0, 0, 100)
gen.add('yaw_ki',     double_t, 0, 'Yaw I',     0, 0, 100)
gen.add('yaw_ki_max', double_t, 0, 'Yaw I Max', 0, 0, 100)
gen.add('yaw_kd',     double_t, 0, 'Yaw D',     0, 0, 100)

exit(gen.generate(PACKAGE, 'attitude_controller', 'attitude_controller_params'))
